aten/src/ATen/native/TensorIndexing.h

```diff
@@ -77,36 +125,454 @@ struct CAFFE2_API TensorIndex final {
+    Tensor result = prev_dim_result.unsqueeze(*dim_ptr);
+    impl::recordTensorIndex(impl::boolToIndexingTensor(result, index.boolean(), original_tensor_device), outIndices, dim_ptr);
+    return result;
+  } else if (index.is_tensor()) {
+    Tensor result = prev_dim_result;
+    const Tensor& tensor = index.tensor();
+    auto scalar_type = tensor.scalar_type();
+    if (tensor.dim() == 0 && at::isIntegralType(scalar_type, /*includeBool=*/true)) {
+      if (scalar_type != at::kByte && scalar_type != at::kBool) {
+        result = impl::applySelect(result, *dim_ptr, tensor.item<int64_t>(), real_dim, original_tensor_device, prev_dim_result_sizes);
+      } else {
+        result = result.unsqueeze(*dim_ptr);
+        if (scalar_type == at::kBool) {
+          impl::recordTensorIndex(impl::boolToIndexingTensor(result, tensor.item<bool>() != 0, original_tensor_device), outIndices, dim_ptr);
+        } else {
+          impl::recordTensorIndex(impl::boolToIndexingTensor(result, tensor.item<uint8_t>() != 0, original_tensor_device), outIndices, dim_ptr);
+        }
+      }
+    } else {
+      impl::recordTensorIndex(tensor, outIndices, dim_ptr);
+    }
+    return result;
+  } else {
+    TORCH_INTERNAL_ASSERT(false, "Invalid TensorIndex type");
+  }
+}
+
+namespace impl {
+// This mirrors `applySlicing` in torch/csrc/autograd/python_variable_indexing.cpp
+static inline Tensor applySlicing(
+    const Tensor& self,
+    const ArrayRef<TensorIndex>& indices,
+    std::vector<Tensor>& outIndices,
+    bool is_tracing,
+    const at::Device& self_device,
+    const IntArrayRef& self_sizes) {
+  int64_t dim = 0;
+  int64_t specified_dims = impl::count_specified_dimensions(indices);
+
+  TORCH_CHECK_INDEX(specified_dims <= self_sizes.size(), "too many indices for tensor of dimension ", (int)self_sizes.size());
+
+  Tensor result = self;
+  for (int64_t i = 0; i < indices.size(); i++) {
+    auto& obj = indices[i];
+    result = handleDimInMultiDimIndexing(
+      /*prev_dim_result=*/result,
+      /*original_tensor=*/self,
+      /*index=*/obj,
+      /*dim=*/&dim,
+      /*specified_dims=*/&specified_dims,
+      /*real_dim=*/i,
+      /*outIndices=*/outIndices,
+      /*is_tracing=*/is_tracing,
+      /*original_tensor_device=*/self_device,
+      /*prev_dim_result_sizes=*/result.sizes());
+  }
+  return result;
+}
+} // namespace impl
+
+static inline Tensor dispatch_index(const Tensor& self, std::vector<Tensor>&& indices) {
+  return self.index(impl::typeConvertIndices(self, std::move(indices)));
+}
+
+static inline Tensor dispatch_index_put_(Tensor& self, std::vector<Tensor>&& indices, const Tensor& value) {
+  return self.index_put_(impl::typeConvertIndices(self, std::move(indices)), value);
+}
+
+// This mirrors `THPVariable_getitem` in torch/csrc/autograd/python_variable_indexing.cpp
+static inline Tensor get_item(const Tensor& self, const ArrayRef<TensorIndex>& indices, bool is_tracing_and_1d_slice_or_Nd = false) {
```

Comment:
I audited every call site here and I didn't see any use of the `1d_slice_or_Nd` sense of `is_tracing_and_1d_slice_or_Nd`

Reply:
But I think that here, the tracing argument is totally unnecessary. Here's why; look at the use site:

```
        /*ensure_view=*/true,
        /*is_tracing=*/is_tracing_and_1d_slice_or_Nd,
```

When `ensure_view` is true, we NEVER directly return self. So the setting of `is_tracing_and_1d_slice_or_Nd` doesn't matter at all; you're always going to make a fresh object.